gstreamer-pbutils/src/discoverer.rs

```diff
@@ -0,0 +1,65 @@
+// Copyright (C) 2018 Thiago Santos <thiagossantos@gmail.com>
+//                    Sebastian Dröge <sebastian@centricular.com>
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use gst;
+
+use Discoverer;
+
+use glib;
+use glib::Value;
+use glib::IsA;
+use glib::signal::connect;
+use glib::signal::SignalHandlerId;
+use glib::translate::*;
+use glib::object::Downcast;
+
+use ffi;
+use glib_ffi;
+use gobject_ffi;
+
+use std::mem::transmute;
+use std::boxed::Box as Box_;
+
+pub trait DiscovererExtManual {
```

Comment:
If you remove the `DiscovererExt` trait, this can also just be an `impl Discoverer`